lib/refile/s3.rb

```diff
@@ -52,7 +46,7 @@ def initialize(access_key_id:, secret_access_key:, region:, bucket:, max_size: n
     verify_uploadable def upload(uploadable)
       id = @hasher.hash(uploadable)
 
-      if uploadable.is_a?(Refile::File) and uploadable.backend.is_a?(S3) and uploadable.backend.access_key_id == access_key_id
```

Comment:
Yes, i think we can fetch access_key_id from client configuration. it will work in all auth cases:
* Credentials passed to the AWS.config (this method now we use)
* Environment Variables
* The credentials file's default profile
* Instance profile credentials (this method i need)

more about this - http://docs.aws.amazon.com/AWSSdkDocsRuby/latest//DeveloperGuide/prog-basics-creds.html)
I fix my PR soon